Apps/Admin/Client/Pages/DashboardPage.razor.cs

```diff
@@ -0,0 +1,345 @@
+    {
+        get
+        {
+            if (this.RecurringUsersResult?.Result != null)
+            {
+               return this.RecurringUsersResult?.Result.TotalRecurringUsers ?? 0;
+            }
+
+            return 0;
+        }
+    }
+
+    private int RatingCount
+    {
+        get
+        {
+            if (this.RatingSummaryResult?.Result != null)
+            {
+                return (from result in this.RatingSummaryResult?.Result
+                        select result.Value).Count();
+            }
+
+            return 0;
+        }
+    }
+
+    private IDictionary<string, int>? RatingSummaryResults
+    {
+        get
+        {
+            if (this.RatingSummaryResult?.Result != null)
+            {
+                return this.RatingSummaryResult?.Result;
+            }
+
+            return new Dictionary<string, int>();
+        }
+    }
+
+    // Tuple<index,ProgressBarvalue,RatingTotal>
+    private List<Tuple<string, int, int>>? RatingSummary
+    {
+        get
+        {
+            var ratingSummary = this.RatingSummaryResults;
+            var results = new List<Tuple<string, int, int>>();
+            if (ratingSummary != null)
+            {
+               for (int i = 5; i >= 1; i--)
+                {
+                    string index = i.ToString(CultureInfo.InvariantCulture);
+                    int ratingTotal = (from value in ratingSummary
+                                      where value.Key == index
+                                      select value).Count();
+
+                    int ratingValue = (from value in ratingSummary
+                                       where value.Key == index
+                                       select value.Value).Sum();
+
+                    int item = this.RatingCount > 0 ? (ratingValue / this.RatingCount) * 100 : 0;
+                    results.Add(Tuple.Create(index, item, ratingTotal));
+                }
+            }
+
+            return results;
+        }
+    }
+
+    private string RatingAverage
+    {
+        get
+        {
+            if (this.RatingSummaryResult?.Result != null)
+            {
+                var totalCount = this.RatingCount;
+
+                var ratingSummary = this.RatingSummaryResults;
+                decimal totalScore = 0M;
+                if (ratingSummary != null)
+                {
+                    totalScore = (from value in ratingSummary
+                                 select Convert.ToInt32(value.Key, CultureInfo.InvariantCulture) * value.Value).Sum();
+
+                    return totalCount != 0 ? (totalScore / totalCount).ToString("0.00", CultureInfo.InvariantCulture) : "N/A";
+                }
+            }
+
+            return "N/A";
+        }
+    }
+
+    /// <inheritdoc/>
+    protected override void OnInitialized()
+    {
+        base.OnInitialized();
+        this.ResetDashboardState();
+        this.LoadDispatchActions();
+    }
+
+    private static List<string> PeriodDatesList()
```

Comment:
Based on the usage, I don't think this needs to be a list - it could be two getters for the default start and end date.